clang/lib/Basic/SourceManager.cpp

```diff
@@ -1064,6 +1064,14 @@ bool SourceManager::isAtStartOfImmediateMacroExpansion(SourceLocation Loc,
   return true;
 }
 
+void SourceManager::setLocationForTimeTrace(SourceLocation Loc,
+                                            llvm::TimeTraceMetadata &M) const {
+  Loc = getExpansionLoc(Loc);
+  M.File = getFilename(Loc);
+  M.Line = getExpansionLineNumber(Loc);
```

Comment:
Suggestion: `getLineNumber` should produce the same result since we've called the `getExpansionLoc` above already.
Will read simpler and should skip some redundant work.